### PR TITLE
WebIDL: Use ArrayBufferViewU8 instead of raw JSObject in bindings

### PR DESCRIPTION
**Pull request by Taym95**
Part #31064, we would like to remove the direct use of `unsafe NonNull<JSObject>`, and replace it with use of the appropriate safe wrapper found in [typedarray.rs](https://github.com/servo/mozjs/blob/de842d84a07eabac4e1bfe72a954b74ee0244b82/mozjs/src/typedarray.rs), in this PR we are replacing `unsafe NonNull<JSObject>` with `ArrayBufferView`, according to [spec](https://github.com/servo/servo/blob/c94d5842db11330ff32b0dd2b8ce036c53b410bb/components/script/dom/webidls/Crypto.webidl#L22) 

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because there is not behavior change

### CODE REVIEW

File: components/script/dom/crypto.rs [hunk 1]
```diff
@@ -58,9 +59,11 @@ impl CryptoMethods for Crypto {
                 return Err(Error::QuotaExceeded);
             }
             self.rng.borrow_mut().fill_bytes(&mut data);
-        }
 
-        unsafe { Ok(NonNull::new_unchecked(*input.underlying_object())) }
+            rooted!(in (*_cx) let mut array = ptr::null_mut::<JSObject>());
+            create_typed_array::<ArrayBufferViewU8>(_cx, &data, array.handle_mut())
```
[the remark this answers is not in the document]
Comment (sebsebmc) [reply components/script/dom/crypto.rs:64]: You need to change that to the 6 argument macro form.

[the remark this answers is not in the document]
Comment (Taym95) [reply components/script/dom/crypto.rs:64]: thanks, yeah makes sense, I will make PR to mozjs first then

**Comment by mrobinson** [comment]
> part https://github.com/servo/servo/issues/31064

Would it be possible to write a full message for this change explaining the context and what part of the work it does. It's incredibly helpful for those reading the repository history.

File: components/script/dom/crypto.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Taym95) [reply components/script/dom/crypto.rs:64]: I will use `TypedArray::from(*input)`  for now

File: components/script/dom/crypto.rs [hunk 2]
```diff
@@ -58,9 +56,11 @@ impl CryptoMethods for Crypto {
                 return Err(Error::QuotaExceeded);
             }
             self.rng.borrow_mut().fill_bytes(&mut data);
+            TypedArray::<ArrayBufferViewU8, *mut JSObject>::from(*unsafe {
+                input.underlying_object()
+            })
+            .map_err(|_| Error::JSFailed)
```
Comment (gterzian) [inline components/script/dom/crypto.rs:62]: Let's do:

```rust
let underlying_object = unsafe { input.underlying_object() };
TypedArray::<ArrayBufferViewU8, *mut JSObject>::from(*underlying_object)
    map_err(|_| Error::JSFailed)
```

**Review by gterzian — APPROVED** [review]
LGTM with one nit. 

I was a bit surprised that we can only create an `ArrayBufferViewU8`, since the underlying type can be of other elements too, but this seems to match [`get_array_type`](https://github.com/servo/mozjs/blob/d1f1519b8a9a57d97e3f623ea119e86c5dcbf524/mozjs/src/typedarray.rs#L441)(which is only implemented on `ArrayBufferViewU8`). cc @jdm

**Review by gterzian — APPROVED** [review]
LGTM, thanks again @Taym95 and I agree we should later look into perhaps hiding that in `typedarrays`...

**Pull request merged**